### PR TITLE
Update slack lambda to use environment variables

### PR DESCRIPTION
**Pull request by crissupb**
Use environment variables instead of inline values in order to update Lambda functions if values change.
Related to issue https://github.com/18F/identity-security-private/issues/1809

### After merge update identity-devops PR https://github.com/18F/identity-devops/pull/3041 with new gitref.

### CODE REVIEW

**Comment by crissupb** [comment]
TF plan
```
# module.main.module.slack_login_events.aws_lambda_function.slack_lambda will be updated in-place
  ~ resource "aws_lambda_function" "slack_lambda" {
        arn                            = "arn:aws:lambda:us-west-2:894947205914:function:snstoslack_login_events"
        description                    = "Sends messages to #login-events Slack channel via SNS subscription."
      ~ filename                       = ".terraform/modules/main.slack_login_events/slack_lambda/lambda_function.zip" -> "../../../../identity-terraform/slack_lambda/lambda_function.zip"
        function_name                  = "snstoslack_login_events"
        handler                        = "lambda_function.lambda_handler"
        id                             = "snstoslack_login_events"
        invoke_arn                     = "arn:aws:apigateway:us-west-2:lambda:path/2015-03-31/functions/arn:aws:lambda:us-west-2:894947205914:function:snstoslack_login_events/invocations"
        last_modified                  = "2020-10-20T20:35:57.801+0000"
        layers                         = []
        memory_size                    = 128
        publish                        = true
      ~ qualified_arn                  = "arn:aws:lambda:us-west-2:894947205914:function:snstoslack_login_events:7" -> (known after apply)
        reserved_concurrent_executions = -1
        role                           = "arn:aws:iam::894947205914:role/snstoslack_login_events20201020203539837600000002"
        runtime                        = "python3.6"
        source_code_hash               = "GgVRi556YD3qnJ16j4nD+jXViohpvC2FfTOnfYc6mZE="
        source_code_size               = 549
        tags                           = {}
        timeout                        = 120
      ~ version                        = "7" -> (known after apply)

      + environment {
          + variables = {
              + "slack_channel"  = "login-events"
              + "slack_icon"     = ":login-dot-gov:"
              + "slack_username" = "SNSToSlack Notifier"
            }
        }

        tracing_config {
            mode = "PassThrough"
        }
    }

  # module.main.module.slack_login_events_use1.aws_lambda_function.slack_lambda will be updated in-place
  ~ resource "aws_lambda_function" "slack_lambda" {
        arn                            = "arn:aws:lambda:us-east-1:894947205914:function:snstoslack_login_events"
        description                    = "Sends messages to #login-events Slack channel via SNS subscription."
      ~ filename                       = ".terraform/modules/main.slack_login_events_use1/slack_lambda/lambda_function.zip" -> "../../../../identity-terraform/slack_lambda/lambda_function.zip"
        function_name                  = "snstoslack_login_events"
        handler                        = "lambda_function.lambda_handler"
        id                             = "snstoslack_login_events"
        invoke_arn                     = "arn:aws:apigateway:us-east-1:lambda:path/2015-03-31/functions/arn:aws:lambda:us-east-1:894947205914:function:snstoslack_login_events/invocations"
        last_modified                  = "2020-10-20T20:35:39.117+0000"
        layers                         = []
        memory_size                    = 128
        publish                        = true
      ~ qualified_arn                  = "arn:aws:lambda:us-east-1:894947205914:function:snstoslack_login_events:1" -> (known after apply)
        reserved_concurrent_executions = -1
        role                           = "arn:aws:iam::894947205914:role/snstoslack_login_events20201020203513188700000001"
        runtime                        = "python3.6"
        source_code_hash               = "GgVRi556YD3qnJ16j4nD+jXViohpvC2FfTOnfYc6mZE="
        source_code_size               = 549
        tags                           = {}
        timeout                        = 120
      ~ version                        = "1" -> (known after apply)

      + environment {
          + variables = {
              + "slack_channel"  = "login-events"
              + "slack_icon"     = ":login-dot-gov:"
              + "slack_username" = "SNSToSlack Notifier"
            }
        }

        tracing_config {
            mode = "PassThrough"
        }
    }

  # module.main.module.slack_login_otherevents.aws_lambda_function.slack_lambda will be updated in-place
  ~ resource "aws_lambda_function" "slack_lambda" {
        arn                            = "arn:aws:lambda:us-west-2:894947205914:function:snstoslack_login_otherevents"
        description                    = "Sends messages to #login-otherevents Slack channel via SNS subscription."
      ~ filename                       = ".terraform/modules/main.slack_login_otherevents/slack_lambda/lambda_function.zip" -> "../../../../identity-terraform/slack_lambda/lambda_function.zip"
        function_name                  = "snstoslack_login_otherevents"
        handler                        = "lambda_function.lambda_handler"
        id                             = "snstoslack_login_otherevents"
        invoke_arn                     = "arn:aws:apigateway:us-west-2:lambda:path/2015-03-31/functions/arn:aws:lambda:us-west-2:894947205914:function:snstoslack_login_otherevents/invocations"
        last_modified                  = "2020-10-20T20:35:57.981+0000"
        layers                         = []
        memory_size                    = 128
        publish                        = true
      ~ qualified_arn                  = "arn:aws:lambda:us-west-2:894947205914:function:snstoslack_login_otherevents:7" -> (known after apply)
        reserved_concurrent_executions = -1
        role                           = "arn:aws:iam::894947205914:role/snstoslack_login_otherevents20201020203539989800000003"
        runtime                        = "python3.6"
        source_code_hash               = "0yXNAPdGoDkohE8m0TYLo1z1aWyFcbX/vkMghBsq1Xg="
        source_code_size               = 552
        tags                           = {}
        timeout                        = 120
      ~ version                        = "7" -> (known after apply)

      + environment {
          + variables = {
              + "slack_channel"  = "login-otherevents"
              + "slack_icon"     = ":login-dot-gov:"
              + "slack_username" = "SNSToSlack Notifier"
            }
        }

        tracing_config {
            mode = "PassThrough"
        }
    }

  # module.main.module.slack_login_otherevents_use1.aws_lambda_function.slack_lambda will be updated in-place
  ~ resource "aws_lambda_function" "slack_lambda" {
        arn                            = "arn:aws:lambda:us-east-1:894947205914:function:snstoslack_login_otherevents"
        description                    = "Sends messages to #login-otherevents Slack channel via SNS subscription."
      ~ filename                       = ".terraform/modules/main.slack_login_otherevents_use1/slack_lambda/lambda_function.zip" -> "../../../../identity-terraform/slack_lambda/lambda_function.zip"
        function_name                  = "snstoslack_login_otherevents"
        handler                        = "lambda_function.lambda_handler"
        id                             = "snstoslack_login_otherevents"
        invoke_arn                     = "arn:aws:apigateway:us-east-1:lambda:path/2015-03-31/functions/arn:aws:lambda:us-east-1:894947205914:function:snstoslack_login_otherevents/invocations"
        last_modified                  = "2020-10-20T20:35:50.108+0000"
        layers                         = []
        memory_size                    = 128
        publish                        = true
      ~ qualified_arn                  = "arn:aws:lambda:us-east-1:894947205914:function:snstoslack_login_otherevents:1" -> (known after apply)
        reserved_concurrent_executions = -1
        role                           = "arn:aws:iam::894947205914:role/snstoslack_login_otherevents20201020203513839200000003"
        runtime                        = "python3.6"
        source_code_hash               = "0yXNAPdGoDkohE8m0TYLo1z1aWyFcbX/vkMghBsq1Xg="
        source_code_size               = 552
        tags                           = {}
        timeout                        = 120
      ~ version                        = "1" -> (known after apply)

      + environment {
          + variables = {
              + "slack_channel"  = "login-otherevents"
              + "slack_icon"     = ":login-dot-gov:"
              + "slack_username" = "SNSToSlack Notifier"
            }
        }

        tracing_config {
            mode = "PassThrough"
        }
    }

  # module.main.module.slack_login_soc.aws_lambda_function.slack_lambda will be updated in-place
  ~ resource "aws_lambda_function" "slack_lambda" {
        arn                            = "arn:aws:lambda:us-west-2:894947205914:function:snstoslack_login_soc"
        description                    = "Sends messages to #login-soc Slack channel via SNS subscription."
      ~ filename                       = ".terraform/modules/main.slack_login_soc/slack_lambda/lambda_function.zip" -> "../../../../identity-terraform/slack_lambda/lambda_function.zip"
        function_name                  = "snstoslack_login_soc"
        handler                        = "lambda_function.lambda_handler"
        id                             = "snstoslack_login_soc"
        invoke_arn                     = "arn:aws:apigateway:us-west-2:lambda:path/2015-03-31/functions/arn:aws:lambda:us-west-2:894947205914:function:snstoslack_login_soc/invocations"
        last_modified                  = "2020-11-05T15:44:42.510+0000"
        layers                         = []
        memory_size                    = 128
        publish                        = true
      ~ qualified_arn                  = "arn:aws:lambda:us-west-2:894947205914:function:snstoslack_login_soc:7" -> (known after apply)
        reserved_concurrent_executions = -1
        role                           = "arn:aws:iam::894947205914:role/snstoslack_login_soc20201020203539546100000001"
        runtime                        = "python3.6"
        source_code_hash               = "EffQ3k7wqRDkPULopPwDTKUrrQhgdVLZvN5dWcrx6aI="
        source_code_size               = 545
        tags                           = {}
        timeout                        = 120
      ~ version                        = "7" -> (known after apply)

      ~ environment {
          ~ variables = {
              ~ "slack_channel"  = "#login-soc-events" -> "login-soc-events"
                "slack_icon"     = ":login-dot-gov:"
                "slack_username" = "SNSToSlack Notifier"
            }
        }

        tracing_config {
            mode = "PassThrough"
        }
    }

  # module.main.module.slack_login_soc_use1.aws_lambda_function.slack_lambda will be updated in-place
  ~ resource "aws_lambda_function" "slack_lambda" {
        arn                            = "arn:aws:lambda:us-east-1:894947205914:function:snstoslack_login_soc"
        description                    = "Sends messages to #login-soc Slack channel via SNS subscription."
      ~ filename                       = ".terraform/modules/main.slack_login_soc_use1/slack_lambda/lambda_function.zip" -> "../../../../identity-terraform/slack_lambda/lambda_function.zip"
        function_name                  = "snstoslack_login_soc"
        handler                        = "lambda_function.lambda_handler"
        id                             = "snstoslack_login_soc"
        invoke_arn                     = "arn:aws:apigateway:us-east-1:lambda:path/2015-03-31/functions/arn:aws:lambda:us-east-1:894947205914:function:snstoslack_login_soc/invocations"
        last_modified                  = "2020-10-20T20:35:44.095+0000"
        layers                         = []
        memory_size                    = 128
        publish                        = true
      ~ qualified_arn                  = "arn:aws:lambda:us-east-1:894947205914:function:snstoslack_login_soc:1" -> (known after apply)
        reserved_concurrent_executions = -1
        role                           = "arn:aws:iam::894947205914:role/snstoslack_login_soc20201020203513189700000002"
        runtime                        = "python3.6"
        source_code_hash               = "ZIB5F830R5oVctWuMsvb86WALs97PShElujhEWp/vms="
        source_code_size               = 548
        tags                           = {}
        timeout                        = 120
      ~ version                        = "1" -> (known after apply)

      + environment {
          + variables = {
              + "slack_channel"  = "login-soc-events"
              + "slack_icon"     = ":login-dot-gov:"
              + "slack_username" = "SNSToSlack Notifier"
            }
        }

        tracing_config {
            mode = "PassThrough"
        }
    }

```

File: slack_lambda/main.tf [hunk 1]
```diff
@@ -64,14 +64,18 @@ locals {
     #!/usr/bin/python3.6
     import urllib3
     import json
+    import os
+    slackChannel = os.environ['slack_channel']
```
Comment (pauldoomgov) [inline slack_lambda/main.tf:68]: This makes sense but can we have it fallback to using TF replacement?
~~~
slackChannel = os.environ.get('slack_channel', "#${var.slack_channel}")
slackUsername = os.environ.get('slack_username', "${var.slack_username}")
slackIcon = os.environ.get('slack_icon', "${var.slack_icon}")
~~~

**Review by pauldoomgov — APPROVED** [review]
I have a suggested change but not worth blocking.   LGTM!

File: slack_lambda/main.tf (hunk 1 above)
Comment (crissupb) [reply slack_lambda/main.tf:68]: Sometimes I thick skull.  Why, it doesn't work the way it was.

Comment (pauldoomgov) [reply slack_lambda/main.tf:68]: Read the whole PR, Paul.   Nevermind.

File: slack_lambda/main.tf [hunk 2]
```diff
@@ -166,6 +170,13 @@ resource "aws_lambda_function" "slack_lambda" {
       last_modified,
     ]
   }
+  environment {
```
Comment (pauldoomgov) [inline slack_lambda/main.tf:173]: Oh!  Riiiiggght :)

**Review by pauldoomgov — APPROVED** [review]
Do it!

**Pull request merged**